src/rhsm/profile.py

```diff
@@ -196,111 +196,28 @@ def collect(self) -> List[dict]:
-        enabled_repos = []
-        for section in enabled_sections:
-            try:
-                enabled_repos.append(
-                    {
-                        "repositoryid": section,
-                        "baseurl": [self._format_baseurl(config.get(section, "baseurl"))],
-                    }
-                )
-            except ImportError:
-                break
-        return enabled_repos
-
-    def __init__(self, repo_file: str) -> None:
-        """
-        Initialize EnabledRepos
-        :param repo_file: A repo file path used to filter the report.
-        """
-        if dnf is not None:
-            self.db = dnf.dnf.Base()
-        elif yum is not None:
-            self.yb = yum.YumBase()
-
-        self.repofile: str = repo_file
-        self.content: List[dict] = self.__generate()
-
-    def __str__(self) -> str:
-        return str(self.content)
-
-    def _format_baseurl(self, repo_url: str) -> str:
-        """
-        Returns a well formatted baseurl string
-        :param repo_url: a repo URL that you want to format
-        """
-        if use_zypper:
-            return self._cut_question_mark(repo_url)
-        else:
-            mappings = self._obtain_mappings()
-            return repo_url.replace("$releasever", mappings["releasever"]).replace(
-                "$basearch", mappings["basearch"]
-            )
-
-    def _cut_question_mark(self, repo_url) -> str:
-        """
-        Returns a string where everything after the first occurrence of '?' is truncated
-        :param repo_url: a repo URL that you want to modify
-        """
-        return repo_url[: repo_url.find("?")]
-
-    @suppress_output
-    def _obtain_mappings(self) -> dict:
-        """
-        returns a hash with "basearch" and "releasever" set. This will try dnf first, and them yum if dnf is
-        not installed.
-        """
-        if dnf is not None:
-            return self._obtain_mappings_dnf()
-        elif yum is not None:
-            return self._obtain_mappings_yum()
-        else:
-            log.error("Unable to load module for any supported package manager (dnf, yum).")
-            raise ImportError
-
-    def _obtain_mappings_dnf(self) -> dict:
-        return {
-            "releasever": self.db.conf.substitutions["releasever"],
-            "basearch": self.db.conf.substitutions["basearch"],
-        }
-
-    def _obtain_mappings_yum(self) -> dict:
-        return {"releasever": self.yb.conf.yumvar["releasever"], "basearch": self.yb.conf.yumvar["basearch"]}
-
-
 class EnabledReposProfile:
     """
     Collect information about enabled repositories
     """
 
     def __init__(self, repo_file: str = REPOSITORY_PATH) -> None:
-        self._enabled_repos: EnabledRepos = EnabledRepos(repo_file)
+        self._content = []
+        for repo_file_cls, _ in get_repo_file_classes():
+            repo_file = repo_file_cls()
```

Comment:
Should this not rather be:
`repo_file = repo_file_cls(repo_file)`
This might have further implications, since apparently for now this expects to always use the `repo_file_cls()`'s default-value for `path`